pkg/inventory/inventory-client.go

```diff
@@ -158,30 +180,36 @@ func (cic *ClusterInventoryClient) Replace(localInv InventoryInfo, objs object.O
 	if err != nil {
 		return fmt.Errorf("failed to read inventory objects from cluster: %w", err)
 	}
-	if objs.Equal(clusterObjs) {
-		klog.V(4).Infof("applied objects same as cluster inventory: do nothing")
-		return nil
-	}
 	clusterInv, err := cic.GetClusterInventoryInfo(localInv, dryRun)
 	if err != nil {
 		return fmt.Errorf("failed to read inventory from cluster: %w", err)
 	}
-	clusterInv, err = cic.replaceInventory(clusterInv, objs)
+	clusterInv, err = cic.replaceInventory(clusterInv, objs, applyStatus)
 	if err != nil {
 		return err
 	}
+	if objs.Equal(clusterObjs) {
+		klog.V(4).Infof("applied objects same as cluster inventory: do nothing")
```

Comment:
This log isn't accurate any more if the status is still being updated.

Reply:
Done.